src/data/polynomial/reverse.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import data.polynomial.degree.basic
+import data.polynomial.erase_lead
+import data.polynomial.degree.trailing_degree
+
+/-!
+# Reverse of a univariate polynomial
+
+The main definition is `reverse`.  Applying `reverse` to a polynomial `f(x)` produces the
+polynomial `x^f.nat_degree * f(1/x)`.
+
+The main result is that `reverse (f * g) = reverse (f) * reverse (g)`, provided the leading
+coefficients of `f` and `g` do not multiply to zero.
+-/
+
+namespace polynomial
+
+open polynomial finsupp finset --erase_lead
+
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+
+namespace rev
+/-- rev_at is a function of two natural variables (N,i).  If i ≤ N, then rev_at N i returns N-i,
+otherwise it returns N.  Essentially, this function is only used for i ≤ N. -/
+def rev_at (N : ℕ) : ℕ → ℕ := λ i : ℕ, ite (i ≤ N) (N-i) i
+
+@[simp] lemma rev_at_invol {N n : ℕ} : rev_at N (rev_at N n) = n :=
+begin
+  unfold rev_at,
+  split_ifs with h j,
+  { exact nat.sub_sub_self h, },
+  { exfalso,
+    apply j,
+    exact nat.sub_le N n, },
+  { refl, },
+end
+
+@[simp] lemma rev_at_small {N n : ℕ} (H : n ≤ N) : rev_at N n = N-n :=
+begin
+  unfold rev_at,
+  split_ifs,
+  refl,
+end
+
+/-- The function `reflect` of a natural number `N` and a polynomial `f`, applies the function
+`rev_at` to the exponents of the terms appearing in the expansion of `f`.  In practice, `reflect`
+is only used when `N` is at least as large as the degree of `f`.  Eventually, it will be used with
+`N` exactly equal to the degree of `f`.  -/
+def reflect : ℕ → polynomial R → polynomial R :=
+ λ N : ℕ, λ f : polynomial R, ⟨ image (rev_at N)  (f.support), λ i : ℕ, f.coeff (rev_at N i),
+ begin
+  intro,
+  rw mem_image,
+  split,
+  { intro h,
+    rcases h with ⟨ a, ha, rfl⟩,
+    rwa [rev_at_invol, ← mem_support_iff_coeff_ne_zero], },
+  { intro,
+    use (rev_at N a),
+    rwa [rev_at_invol, eq_self_iff_true, and_true, mem_support_iff_coeff_ne_zero], },
+end ⟩
+
+@[simp] lemma reflect_zero {n : ℕ} : reflect n (0 : polynomial R) = 0 := rfl
+
+@[simp] lemma reflect_zero_iff {N : ℕ} (f : polynomial R) :
+ reflect N (f : polynomial R) = 0 ↔ f=0 :=
```

Comment:
I made ```f``` implicit.  I also added the spacing that you suggested.  I am not sure about the "indentation for continuation lines for types".  I took this to mean: when I break a line because it is longer than 100 chars, I will use two spaces for the continuation, rather than only one.  Let me know if you had something else in mind!